test/sequential/market.nonTransferable.multiswap.test.ts

```diff
@@ -0,0 +1,294 @@
+  await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      api.tx.foundationMembers.changeKey(oldFounder2),
+    ),
+  ).then((events) => {
+    const res = getEventResultFromMangataTx(events);
+    expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+});
+
+test("User can buy GASP in multiswap operation", async () => {
+  const userBalanceBeforeSwap = await api.query.tokens.accounts(
+    testUser,
+    token1,
+  );
+  await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.multiswapAssetSell(
+        [liqIds[0], liqIds[1]],
+        token1,
+        BN_TEN_THOUSAND,
+        GASP_ASSET_ID,
+        BN_ZERO,
+      ),
+    ),
+  ).then(async (result) => {
+    await waitSudoOperationSuccess(result, "SudoAsDone");
+  });
+  const userBalanceAfterSwap = await api.query.tokens.accounts(
+    testUser,
+    token1,
+  );
+  expect(userBalanceAfterSwap.free).bnEqual(
+    userBalanceBeforeSwap.free.sub(BN_TEN_THOUSAND),
+  );
+});
+
+test("User can't sell GASP in multiswap operation (GASP token at the beginning)", async () => {
+  await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.multiswapAssetSell(
+        [liqIds[1], liqIds[0]],
+        GASP_ASSET_ID,
+        BN_TEN_THOUSAND,
+        token1,
+        BN_ZERO,
+      ),
+    ),
+  ).then(async (result) => {
+    await waitSudoOperationFail(result, ["NontransferableToken"], "SudoAsDone");
+  });
+});
+
+test("User can't sell GASP in multiswap operation (GASP token in the middle)", async () => {
+  await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.multiswapAssetSell(
+        [liqIds[1], liqIds[2]],
+        token2,
+        BN_TEN_THOUSAND,
+        token3,
+        BN_ZERO,
+      ),
+    ),
+  ).then(async (result) => {
+    await waitSudoOperationFail(result, ["NontransferableToken"], "SudoAsDone");
+  });
+});
+
+test("Happy path - multiswap with only stable pools", async () => {
+  const userBalance1BeforeSwap = await api.query.tokens.accounts(
+    testUser,
+    token3,
+  );
+  const userBalance2BeforeSwap = await api.query.tokens.accounts(
+    testUser,
+    token5,
+  );
+  await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.multiswapAssetSell(
+        [liqIds[3], liqIds[4]],
+        token3,
+        BN_TEN_THOUSAND,
+        token5,
+        BN_ZERO,
+      ),
+    ),
+  ).then(async (result) => {
+    await waitSudoOperationSuccess(result, "SudoAsDone");
+  });
+  const userBalance1AfterSwap = await api.query.tokens.accounts(
+    testUser,
+    token3,
+  );
+  const userBalance2AfterSwap = await api.query.tokens.accounts(
+    testUser,
+    token5,
+  );
+  expect(userBalance1AfterSwap.free).bnEqual(
+    userBalance1BeforeSwap.free.sub(BN_TEN_THOUSAND),
+  );
+  expect(userBalance2AfterSwap.free).bnGt(userBalance2BeforeSwap.free);
```

Comment:
use the rpc for calculate chained swaps, so you can assert the values more precisely.